x/concentrated-liquidity/types/keys.go

```diff
@@ -67,11 +67,13 @@ func KeyTickPrefix(poolId uint64) []byte {
 	return key
 }
 
-// KeyFullPosition uses pool Id, owner, lower tick, upper tick, and frozenUntil for keys
-func KeyFullPosition(poolId uint64, addr sdk.AccAddress, lowerTick, upperTick int64, frozenUntil time.Time) []byte {
-	frozenUntilKey := osmoutils.FormatTimeString(frozenUntil)
+// KeyFullPosition uses pool Id, owner, lower tick, upper tick, joinTime and freezeDuration for keys
+func KeyFullPosition(poolId uint64, addr sdk.AccAddress, lowerTick, upperTick int64, joinTime time.Time, freezeDuration time.Duration) []byte {
+	joinTimeKey := osmoutils.FormatTimeString(joinTime)
```

Comment:
We can treat this as a `uint64` as we do with incentive records: https://github.com/osmosis-labs/osmosis/blob/6014a7c6f7f330869469f6ffc15f62cff3b854c1/x/concentrated-liquidity/types/keys.go#L92